src/umpire/DeviceAllocator.cpp

```diff
@@ -73,7 +73,8 @@ __device__ void* DeviceAllocator::allocate(size_t size)
 {
   std::size_t counter = atomicAdd(m_counter, size);
   if (*m_counter > m_size) {
-    UMPIRE_ERROR(out_of_memory_error, "DeviceAllocator out of space");
+    // TODO
+    // UMPIRE_ERROR(out_of_memory_error, "DeviceAllocator out of space");
```

Comment:
?

Reply:
Fuuuuudge - I forgot about this. There is something that the rocm compiler does not like about using UMPIRE_ERROR inside a device function. I get these errors when I put it back in:
`error: no matching constructor for initialization of 'std::string' (aka 'basic_string<char>')`
`error: no matching constructor for initialization of 'umpire::out_of_memory_error'`
`error: no matching constructor for initialization of 'std::ostringstream' (aka 'basic_ostringstream<char>')`
`error: invalid operands to binary expression`

Reply:
Could we keep the `UMPIRE_ERROR` for cuda and find another way to error out with rocm?

Reply:
....several hours and much collaboration later...
We (David) discovered that we have to do `#if defined(__HIP_DEVICE_COMPILE__)` before the call to `UMPIRE_ERROR` in the `allocate` function. Something about the parsing that clang does and how the host and device code was not valid in both separate cases...